staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered.go

```diff
@@ -245,37 +245,40 @@ func (b *bufferedBackend) processEvents(events []*auditinternal.Event) {
 	}()
 }
 
-func (b *bufferedBackend) ProcessEvents(ev ...*auditinternal.Event) {
+func (b *bufferedBackend) ProcessEvents(ev ...*auditinternal.Event) bool {
 	// The following mechanism is in place to support the situation when audit
 	// events are still coming after the backend was stopped.
 	var sendErr error
 	var evIndex int
 
-	// If the delegateBackend was shutdown and the buffer channel was closed, an
-	// attempt to add an event to it will result in panic that we should
-	// recover from.
-	defer func() {
-		if err := recover(); err != nil {
-			sendErr = fmt.Errorf("audit backend shut down")
-		}
-		if sendErr != nil {
-			audit.HandlePluginError(PluginName, sendErr, ev[evIndex:]...)
-		}
-	}()
-
-	for i, e := range ev {
-		evIndex = i
-		// Per the audit.Backend interface these events are reused after being
-		// sent to the Sink. Deep copy and send the copy to the queue.
-		event := e.DeepCopy()
+	func () {
+		// If the delegateBackend was shutdown and the buffer channel was closed, an
+		// attempt to add an event to it will result in panic that we should
+		// recover from.
+		defer func() {
+			if err := recover(); err != nil {
+				sendErr = fmt.Errorf("audit backend shut down")
+			}
+			if sendErr != nil {
+				audit.HandlePluginError(PluginName, sendErr, ev[evIndex:]...)
+			}
+		}()
 
-		select {
-		case b.buffer <- event:
-		default:
-			sendErr = fmt.Errorf("audit buffer queue blocked")
-			return
+		for i, e := range ev {
+			evIndex = i
+			// Per the audit.Backend interface these events are reused after being
+			// sent to the Sink. Deep copy and send the copy to the queue.
+			event := e.DeepCopy()
+
+			select {
+			case b.buffer <- event:
+			default:
+				sendErr = fmt.Errorf("audit buffer queue blocked")
+				return
+			}
 		}
-	}
+	}()
+	return sendErr == nil
```

Comment:
Saw it, return false when the buffer is full.

Reply:
Actually, changed to true since buffered backend will not be used when blocking-strict is selected.